spyder/api/preferences.py

```diff
@@ -16,7 +16,10 @@
 from spyder.config.manager import CONF
 from spyder.config.types import ConfigurationKey
 from spyder.api.utils import PrefixedTuple
-from spyder.plugins.preferences.api import SpyderConfigPage, BaseConfigTab
+from spyder.plugins.preferences.widgets.config_widgets import (
+    SpyderConfigPage,
+    BaseConfigTab
+)
```

Comment:
> Although those definitions are widgets, it is also true that they are part of the preferences plugin API, no?

I'd say that not really. I mean, I think what we should add in `spyder/plugins/myplugin/api.py` are things that other plugins need to import to work in relationship to `myplugin`. That's why we have mainly enums for actions in the `api` module of most of our plugins (to place other actions in menus or toolbars before or after them, for instance).

In this particular case, other plugins don't need to import `SpyderConfigPage` and `BaseConfigTab` but `SpyderPreferencesTab` and `PluginConfigPage`, which are declared here.

However, your comment made me realize that the `MOST_IMPORTANT_PAGES` constant that I added to `spyder.plugins.preferences.api` shouldn't be there. That's because that constant is not needed by other plugins but by several modules in the Preferences plugin. So, I put it in `api.py` to avoid circular imports, but I think it's better placed in `spyder.plugins.preferences.__init__`, which is where I'll move it in my next commit.